frontend/src/routes/[[lang=locale]]/(voters)/+layout.svelte

```diff
@@ -1,14 +1,81 @@
 <script lang="ts">
   import {onDestroy} from 'svelte';
+  import {appType} from '$lib/stores';
+  import {t} from '$lib/i18n';
   import {afterNavigate, onNavigate} from '$app/navigation';
   import {settings, showFeedbackPopup, showSurveyPopup, userPreferences} from '$lib/stores';
   import {startPageview, submitAllEvents} from '$lib/utils/analytics/track';
   import {DataConsentPopup} from '$lib/components/dataConsent/popup';
   import {FeedbackPopup} from '$lib/components/feedback/popup';
   import {SurveyPopup} from '$lib/components/survey/popup';
+  import type {BasicPageProps} from '$lib/templates/basicPage';
+  import {startEvent} from '$lib/utils/analytics/track';
+  import {NavItem, setNavigationContext} from '$lib/components/navigation';
+  import {CandidateNav, VoterNav} from '$lib/templates/page/parts';
+  import {setTopBarActionsConext} from '../topBarActions.context';
+  import Header from './Header.svelte';
+  import {setTopBarContext} from '../topBar.context';
+  import {writable} from 'svelte/store';
+  import {tweened} from 'svelte/motion';
+  import {cubicOut} from 'svelte/easing';
+  import {setTopBarProgressContext} from '../topBarProgress.context';
+  import {getPageStylesContext, resetPageStylesContext, setPageStylesContext} from '../context';
+  import {navigating} from '$app/stores';
+
+  export let drawerToggleId: BasicPageProps['drawerToggleId'] = 'pageDrawerToggle';
+  export let mainId: BasicPageProps['mainId'] = 'mainContent';
+  export let navId: BasicPageProps['navId'] = 'pageNav';
+  export let drawerContentClass: BasicPageProps['drawerContentClass'] = undefined;
 
   let doShowFeedbackPopup = false;
   let doShowSurveyPopup = false;
+  let screenWidth = 0;
+
+  let drawerOpen = false;
+  let drawerOpenElement: HTMLButtonElement | undefined;
+
+  /**
+   * Open the drawer. We also focus on the relevant element to make it easy
+   * to toggle it back when using keyboard navigation.
+   */
+  function openDrawer() {
+    drawerOpen = true;
+    // We need a small timeout for drawerCloseButton to be focusable
+    setTimeout(() => document.getElementById('drawerCloseButton')?.focus(), 50);
+    startEvent('menu_open');
+  }
+
+  /**
+   * Close the drawer. We also focus on the relevant element to make it easy
+   * to toggle it back when using keyboard navigation.
+   */
+  function closeDrawer() {
+    drawerOpen = false;
+    drawerOpenElement?.focus();
+  }
+
+  /**
+   * Create a context which can be used by the nested navigation components to close the drawer
+   */
+  setNavigationContext({close: closeDrawer});
+
+  /**
+   * Create contexts for top application bar
+   */
+  setTopBarContext();
+  setTopBarActionsConext();
+  setTopBarProgressContext({
+    max: writable(0),
+    current: tweened(0, {
+      duration: 400,
+      easing: cubicOut
+    })
+  });
+  setPageStylesContext();
+
+  $: if ($navigating) resetPageStylesContext();
```

Comment:
To me it seems `onDestroy` is called rather consistently, but did you find some info stating otherwise? If we can leverage that, we could use it pretty easily make all page-style and header changes specific to the scope (with support for both layouts and pages).

One possibility would be even to make passing the component’s `onDestroy` handler a prerequisite for accessing the context (I think it'd be best to combine the various contexts into one anyway), e.g.:

```ts
// In app.context.ts
// This would be a bit more involved but it'd make life easy on the consumer side

// Push any merged styles to this stack
const styleStack: Array<PageStyles> = [ DEFAULT_SETTINGS ];

// Merge styles and add to stack
function setPageStyles(styles: DeepPartial<PageStyles> = {}): void {
  const current = styleStack[styleStack.length - 1];
  const newStyle = lodash.merge(lodash.cloneDeep(current), lodash.cloneDeep(styles));
  styleStack.push(newStyle);
  // Apply the style here
}

// Use this to move back to a previous state
function resetStack(index: number): void {
  styleStack.splice(index + 1);
  // Apply the style at styleStack[index]
}

// The onDestroy is used to reset the stack to where it was
export function getAppContext(onDestroy: (fn: () => unknown) => void): AppContext {
  const currentIndex = styleStack.length - 1;
  onDestroy(() => resetStack(currentIndex));
  return { setPageStyles }; // And other functions, stores etc.
}

// On a consuming +page/layout.svelte

import { onDestroy } from 'svelte';
import { getAppContext } from '$lib/contexts/app';

const setPageStyles = getAppContext(onDestroy);
setPageStyles({ drawer: { background: 'bg-base-300' } });
// These will reset to the previous value when unmounted
```